src/pkcs15init/pkcs15-myeid.c

```diff
@@ -349,19 +352,33 @@ myeid_new_file(sc_profile_t *profile, sc_card_t *card,
 	sc_file_t *file;
 	sc_path_t *p;
 	char name[64];
-	const char *tag;
+	const char *tag = NULL;
 	int r;
 
 	LOG_FUNC_CALLED(card->ctx);
-	if (type == SC_PKCS15_TYPE_PRKEY_RSA || type == SC_PKCS15_TYPE_PRKEY_EC)
+	switch (type) {
+	case SC_PKCS15_TYPE_PRKEY_RSA:
+	case SC_PKCS15_TYPE_PRKEY_EC:
```

Comment:
coverity should not report these (http://security.coverity.com/blog/2013/Sep/gimme-a-break.html), no need to add comments.